diff --git a/fs/ntfs/super.c b/fs/ntfs/super.c
--- a/fs/ntfs/super.c
+++ b/fs/ntfs/super.c
@@ -91,7 +91,7 @@ static const struct fs_parameter_spec ntfs_parameters[] = {
 	fsparam_flag("sys_immutable",		Opt_sys_immutable),
 	fsparam_flag("nohidden",		Opt_nohidden),
 	fsparam_flag("hide_dot_files",		Opt_hide_dot_files),
-	fsparam_flag("windows_names",		Opt_check_windows_names),
+	fsparam_bool("windows_names",		Opt_check_windows_names),
 	fsparam_flag("acl",			Opt_acl),
 	fsparam_flag("discard",			Opt_discard),
 	fsparam_flag("sparse",			Opt_sparse),

