diff --git a/acer-wmi-ext.c b/acer-wmi-ext.c
index 24112bf..7829c19 100644
--- a/acer-wmi-ext.c
+++ b/acer-wmi-ext.c
@@ -69,9 +69,9 @@ struct set_battery_health_control_output {
 
 enum battery_mode { HEALTH_MODE = 1, CALIBRATION_MODE = 2 };
 
-#define ACER_SYSTEM_CONTROL_MODE_EC_OFFSET 0x45
+#define ACER_SYSTEM_CONTROL_MODE_EC_OFFSET 0x10
 enum system_control_mode {
-	   	SYSTEM_CONTROL_BALANCED = 1,
+	   	SYSTEM_CONTROL_BALANCED = 0,
 		SYSTEM_CONTROL_SILENT = 2,
 		SYSTEM_CONTROL_PERFORMANCE = 3,
 };
@@ -146,9 +146,9 @@ static struct quirk_entry quirk_unknown = {
 static struct quirk_entry quirk_acer_system_control_mode = {
 	.system_control_mode = 1,
 };
-static struct quirk_entry quirk_acer_sfg174_73 = {
+static struct quirk_entry quirk_acer_sfg14_63 = {
 	.system_control_mode = 1, // Enable system control mode for this model
-	.usb_charge_mode = 1, // Enable USB charge mode for this model
+	.usb_charge_mode = 0, // Disable USB charge mode for this model
 };
 
  /*
@@ -165,12 +165,12 @@ static int __init dmi_matched(const struct dmi_system_id *dmi)
 static const struct dmi_system_id acer_quirks[] __initconst = {
 	{
 		.callback = dmi_matched,
-		.ident = "Acer Swift SFG14-73",
+		.ident = "Acer Swift SFG14-63",
 		.matches = {
 			DMI_MATCH(DMI_SYS_VENDOR, "Acer"),
-			DMI_MATCH(DMI_PRODUCT_NAME, "Swift SFG14-73"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Swift SFG14-63"),
 		},
-		.driver_data = &quirk_acer_sfg174_73,
+		.driver_data = &quirk_acer_sfg14_63,
 	},
 };
 
